Documentation/Commands/strings.md

```diff
@@ -46,3 +46,170 @@ This parser currently doesn't accept any options.
+        <string>NSStringPluralRuleType</string>
+        <key>NSStringFormatValueTypeKey</key>
+        <string>ld</string>
+        <key>zero</key>
+        <string>No matches</string>
+        <key>one</key>
+        <string>%ld match</string>
+        <key>other</key>
+        <string>%ld matches</string>
+    </dict>
+</dict>
+```
+
+</details>
+
+<details>
+
+<summary>Multiple variables with different types in format key</summary>
+
+```xml
+<key>multiple.placeholders-and-variables.string-int</key>
+<dict>
+    <key>NSStringLocalizedFormatKey</key>
+    <string>%#@element@ %#@has_rating@</string>
+    <key>element</key>
+    <dict>
+        <key>NSStringFormatSpecTypeKey</key>
+        <string>NSStringPluralRuleType</string>
+        <key>NSStringFormatValueTypeKey</key>
+        <string>@</string>
+        <key>other</key>
+        <string>%@</string>
+    </dict>
+        <key>has_rating</key>
+    <dict>
+        <key>NSStringFormatSpecTypeKey</key>
+        <string>NSStringPluralRuleType</string>
+        <key>NSStringFormatValueTypeKey</key>
+        <string>d</string>
+        <key>zero</key>
+        <string>has no rating</string>
+        <key>one</key>
+        <string>has one rating</string>
+        <key>other</key>
+        <string>has %d ratings</string>
+    </dict>
+</dict>
+```
+
+</details>
+
+<details>
+
+<summary>Variables with positional arguments in format key</summary>
+
+_Note:_ To have the correct sorting of the parameters of the generated method, in this case `Int, String, String`, it is required to use matching positional arguments for the format key and the placeholder in the rules of the variable.
+
+```xml
+<key>multiple.placeholders-and-variables.int-string-string</key>
+<dict>
+    <key>NSStringLocalizedFormatKey</key>
+    <string>Your %3$#@third@ list contains %1$#@first@ %2$#@second@.</string>
+    <key>first</key>
+    <dict>
+        <key>NSStringFormatSpecTypeKey</key>
+        <string>NSStringPluralRuleType</string>
+        <key>NSStringFormatValueTypeKey</key>
+        <string>d</string>
+        <key>zero</key>
+        <string>no items.</string>
+        <key>one</key>
+        <string>one item. You should buy it</string>
+        <key>other</key>
+        <string>%1$d items. You should buy them</string>
+    </dict>
+    <key>second</key>
+    <dict>
+        <key>NSStringFormatSpecTypeKey</key>
+        <string>NSStringPluralRuleType</string>
+        <key>NSStringFormatValueTypeKey</key>
+        <string>@</string>
+        <key>other</key>
+        <string>%2$@</string>
+    </dict>
+    <key>third</key>
+    <dict>
+        <key>NSStringFormatSpecTypeKey</key>
+        <string>NSStringPluralRuleType</string>
+        <key>NSStringFormatValueTypeKey</key>
+        <string>@</string>
+        <key>other</key>
+        <string>%3$@</string>
+    </dict>
+</dict>
+```
+
+</details>
+
+### Not supported
+
+<details>
+
+<summary>Nested format keys in variables</summary>
+
```

Comment:
```suggestion

Note: in practice this should hopefully be very rare. Especially, if you're using tools like Lokalize, PhraseApp, POEditor, or similar to export your `stringsdict`, it's unlikely that they'll ever generate that kind of convoluted structure for your `stringsdict`.

```

Reply:
Wondering if it's worth also mentioning the workaround if a user ever encounter that need / case?

The workaround being what we currently suggest in the current version where we don't yet support `.stringsdict`, i.e.:
>  - Declare the same key in your `.strings` file too (in addition to your actual entry in your `.stringsdict` with the actual translation that you want to use at runtime) and in that entry in your `.strings` file, give it a (potentially dummy) value that contains all the placeholders you expect for your parameters
>  - Force the `filter` for the `strings` entry in your config file to be `[^/]\.strings$` so that SwiftGen only parses `.strings` files but doesn't parse `.stringsdict` files anymore for your case.
> 
> With that setup, contrary to the default behaviour, SwiftGen will now rely only on the keys in the `.strings` file giving you more control on the generated code when you have cases in your `.stringsdict` file that SwiftGen doesn't support yet

Reply:
But is this a workaround for nested plurals? I don't know how this should work, as the key declared in the `.strings` file would also need to know the correct amount / order of placeholders beforehand, which is not always the same for the nested plurals

Reply:
The point of the workaround is that since the key in the `.strings` file is ignored by Foundation at runtime (when the same key is in `.stringsdict` you can actually put whatever you want as the value in your `.strings` file. And especially, if you configure SwiftGen to only parse the `.strings` file and not the `.stringsdict` file, you can put whatever string with the placeholders you need to make SwiftGen generate the parameters you think are right for this key.

Sure, that would mean that you'll lose the type-safety guarantee provided by SwiftGen (as in: since you can put whatever string you want there so that SwiftGen generates the type of parameters you want, if you put something that doesn't match the actual parameters expected at runtime given the value of `.stringsdict`, you'll crash, and it's now your responsibility to keep your `.strings` fake value with the custom placeholders you want to force here in sync with the actual placeholders you think are expected at runtime by the plural, potentially nested, plural value.

But the point is: the need for this workaround should hopefully not happen in the first place anyway, and at least if someone encounters that case, they have a workaround to continue using SwiftGen in the rest of their project in the meantime, until we fix it and release a new version